packages/miniflare/test/plugins/r2/index.spec.ts

```diff
@@ -1237,31 +1227,16 @@ test("completeMultipartUpload", async (t) => {
 	objectBody = await r2.get("key");
 	t.is(await objectBody?.text(), "1");
 
-	// Check completing with overridden part
-	const upload3 = await r2.createMultipartUpload("key");
-	let part1a = await upload3.uploadPart(1, "value");
-	let part1b = await upload3.uploadPart(1, "value");
-	t.is(part1a.partNumber, part1b.partNumber);
-	t.not(part1a.etag, part1b.etag);
+	// Check ETag is valid MD5
+	const upload4 = await r2.createMultipartUpload("key");
+	part1 = await upload4.uploadPart(1, "1".repeat(PART_SIZE));
+	t.is(part1.etag, "11a36665548da6e3a6336e69cf565d11");
+
 	const notFoundExpectations: ThrowsExpectation<Error> = {
 		instanceOf: Error,
 		message:
 			"completeMultipartUpload: One or more of the specified parts could not be found. (10025)",
 	};
-	await t.throwsAsync(upload3.complete([part1a]), notFoundExpectations);
-	object = await upload3.complete([part1b]);
-	t.is(object.size, 5);
-
-	// Check completing with multiple parts of same part number
-	const upload4 = await r2.createMultipartUpload("key");
-	part1a = await upload4.uploadPart(1, "1".repeat(PART_SIZE));
-	part1b = await upload4.uploadPart(1, "2".repeat(PART_SIZE));
-	const part1c = await upload4.uploadPart(1, "3".repeat(PART_SIZE));
-	await t.throwsAsync(
-		upload4.complete([part1a, part1b, part1c]),
-		internalErrorExpectations("completeMultipartUpload")
-	);
-
```

Comment:
Why are these tests being removed?

Reply:
Just re-added one test that on a second pass, would still pass.

The first one in your selection here was removed because it relies on the test above to attempt to upload the same data multiple times and get different ETags. Since ETags are now the same across multiple uploadParts of the same content, it doesn't make sense to test that case.